src/Microsoft.ML.Core/Prediction/IPredictor.cs

```diff
@@ -12,8 +12,7 @@ namespace Microsoft.ML
     /// <see cref="ITrainer"/> and <see cref="IPredictor"/> it is still useful, but for things based on
     /// the <see cref="IEstimator{TTransformer}"/> idiom, it is inappropriate.
     /// </summary>
-    [BestFriend]
-    internal enum PredictionKind
+    public enum PredictionKind
```

Comment:
These types aren't intended to be public. Can you keep them internal?

Reply:
Unfortunately not if we want the `OneVersusAllModelParameters` to be strongly typed and keep the backwards api compatibility. Since the original type was of `IPredictorProducing<float>`, once I type it that way then `IPredictorProducing<T>`, `IPredictor`, and `PredictionKind` have to all become public as well. The only other thought I had was that maybe I could change the type of the backwards compatible `OneVersusAllModelParameters` to be of type `object` instead of `IPredictorProducing<T>`. This will still keep the backwards API compatibility and would allow me to make those classes internal again.

Reply:
Another option would be to not have the base class be generic. Have `class OneVersusAllModelParametersBase` and `class OneVersusAllModelParameters : OneVersusAllModelParametersBase` and `class OneVersusAllModelParameters<T> : OneVersusAllModelParametersBase`.

These IPredictor types are not meant to be public. We explicitly made them internal in `1.0` because they shouldn't be exposed to the user. So we will have to find another way here.

Reply:
I thought about that approach initially, but there is enough Strongly typed parameters in there I didn't think it would work. Let me test it and see what I can do there. If not, we can always change the type to `object` since that will work and not change anything about the api.

Reply:
Alright, I have made the change following your suggestions, so now the interfaces are back to being internal. Thanks for your feedback!

Reply:
`PredictionKind` should remain internal.